lib/vagrant-service-manager/service_base.rb

```diff
@@ -17,6 +17,43 @@ def service_start_allowed?
       def cdk?
         @machine.guest.capability(:os_variant) == 'cdk'
       end
+
+      def proxy_cmd_options
+        options = ''
+
+        return options unless validation_success?
+
+        PROXY_CONFIG.each do |key|
+          options += "#{key.to_s.upcase}='#{@machine.config.servicemanager.send(key)}' "
+        end
+
+        options.chop
+      end
+
+      def validation_success?
+        proxy = @machine.config.servicemanager.send('http_proxy')
+        user = @machine.config.servicemanager.send('http_proxy_user')
+        password = @machine.config.servicemanager.send('http_proxy_password')
+
+        if proxy && user.nil? && password.nil? || (proxy && user && password)
+          PluginLogger.debug('Detected proxy settings. Going to apply them to service commands.')
+          values = "Proxy URL : #{proxy}"
+          values += ", User : #{user}, Password: #{password}" unless user.nil?
```

Comment:
I like it. However, I think we should not echo the password plain text. The settings might come via environment variables to avoid displaying the password anywhere. Yet this would allow to dump it onto the console. Maybe just "***"!?